### PR TITLE
[doc] docker quick start flink

### LINKED ISSUE #1151: [Docs] Support the Quick Start with Docker in engines document
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Document provides the [Quick Start](https://paimon.apache.org/docs/master/engines/flink/#quick-start) to help users to use Paimon locally. We could support the Quick Start with Docker to help users to use Paimon in Docker/K8S environment.

### Solution

Document supports the Quick Start with Docker to help users to use Paimon in Docker/K8S environment.

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by FireFreedomK**
### Purpose

Linked issue:  #1151 

Add a docker folder in the project root path to store docker compose file. In this PR, a new flink_docker_compose.yml file is added to quickly start the Flink cluster, and paimon-related jar files are introduced. A flink SQL-client container is started with flink_docker_compose.yml to execute the relevant SQL statements to create the catalog and tables, write and query the data.

In the future, add files such as spark3_docker_compose.yml and hive_docker_compose.yml in the docker/ path to quickly build related cluster environments.

New quick_start.md file under docs/content/engines/ that describes how to quickly build flink and paimon environments and how to use paimon through Flink sql-client.

Follow-up quick-start documents such as Spark and Hive will be appended to the file.

### Tests
no

### API and Format
no

### Documentation
no

### CODE REVIEW

File: docs/content/engines/quick_start.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+# Qucik Start
+This document explains how to use Docker to quickly build a 
+practice environment for Paimon and big data engines (Flink,
+Spark, Hive, etc.).
+
+The following steps have been tested on a Windows 11 laptop 
+with docker desktop installed. 
+
+## Prerequisites
+windows 11
```
Comment (SteNicholas) [inline docs/content/engines/quick_start.md:37]: Why the prerequisites are windows 11? Could not other os system support?

Comment (FireFreedomK) [reply docs/content/engines/quick_start.md]: I have a windows 11 PC and other systems should be supported. Just follow docker, docker compose

Comment (FireFreedomK) [reply docs/content/engines/quick_start.md]: In my most recent submission, I added that the answer is yes, but I don't have the right Linux/MacOS computer at hand

File: docker/flink_docker_compose.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+version: "2.2"
```
Comment (JingsongLi) [inline docker/flink_docker_compose.yml:1]: remove this file? Its content has already appeared in the document

File: docs/content/engines/quick_start.md [hunk 3]
```diff
@@ -0,0 +1,332 @@
+---
+title: "Quick Start"
```
Comment (JingsongLi) [inline docs/content/engines/quick_start.md:2]: Can you create a new First level directory named `docker`?

File: docs/content/engines/quick_start.md [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+    user: flink
+    environment:
+      - |
+        FLINK_PROPERTIES=
+        jobmanager.rpc.address: jobmanager
+    networks:
+      - flink-network
+    volumes:
+      - ./custom_jars/flink-shaded-hadoop-2-uber-2.8.3-10.0.jar:/opt/flink/lib/flink-shaded-hadoop-2-uber-2.8.3-10.0.jar
+      - ./custom_jars/paimon-flink-1.17-0.5-20230815.001819-123.jar:/opt/flink/lib/paimon-flink-1.17-0.5-20230815.001819-123.jar
```
Comment (JingsongLi) [inline docs/content/engines/quick_start.md:82]: Can here be `paimon-flink-1.17-{{< version >}}.jar`?

**Comment by pengmide** [comment]
Is this PR still in progress? @FireFreedomK @SteNicholas @JingsongLi

**Comment by JingsongLi** [comment]
Hi @pengmide , it looks like there is no enough requirement for docker quickstart now.

**Pull request closed without merging**